src/app/(proper_react)/redesign/(authenticated)/user/dashboard/subscribed/page.tsx

```diff
@@ -2,76 +2,45 @@
-import { headers } from "next/headers";
-
-export default async function Subscribed() {
-  const session = await getServerSession(authOptions);
-  if (!session?.user?.subscriber) {
-    throw new Error("No session");
-  }
-
-  if (!(await isEligibleForPremium(session.user, getCountryCode(headers())))) {
-    throw new Error("Not eligible for premium");
-  }
-
-  const result = await getOnerepProfileId(session.user.subscriber.id);
-  const profileId = result[0]["onerep_profile_id"] as number;
-
-  try {
-    await activateProfile(profileId);
-    await optoutProfile(profileId);
-  } catch (ex) {
-    console.debug(ex); // TODO handle
-  }
-
-  const dev =
-    process.env.NODE_ENV === "development" || process.env.APP_ENV === "heroku";
-  const latestScan = await getLatestOnerepScanResults(profileId);
-  if (!latestScan.scan) {
-    throw new Error("Must have performed manual scan");
-  }
-
-  const scans = await getAllScanResults(profileId);
-
-  // In dev mode, record scans every time this page is reloaded.
-  // The webhoook does this in production.
-  if (dev) {
-    await addOnerepScanResults(
-      profileId,
-      latestScan.scan.onerep_scan_id,
-      scans,
-      "initial"
-    );
-  }
-
-  return (
-    <div>
-      <div>
-        <h3>You are now subscribed</h3>
-        {dev ? (
-          <div>
-            <h3>Dev mode enabled</h3>
-            <p>Reload this page to update scan</p>
-            <pre>{JSON.stringify(scans, null, 2)}</pre>
-          </div>
-        ) : (
-          ""
-        )}
-      </div>
-    </div>
-  );
+"use client";
+
+import { useSession } from "next-auth/react";
+import { useEffect } from "react";
+import { hasPremium } from "../../../../../../functions/universal/user";
+import { captureException } from "@sentry/browser";
+
+/**
+ * Client-side page to update session info.
+ *
+ * Next-Auth does not have a simple way to do this purely from the client-side, so we
+ * use this page to check and redirect appropriately.
+ *
+ * NOTE: this does not replace doing server-side `hasPremium` checks! This is just
+ * a convenience so users do not need to sign out and back in to refresh their session
+ * after subscribing.
+ */
+export default function Subscribed() {
+  const { update } = useSession();
+
+  useEffect(() => {
+    async function updateSession() {
+      // You can await here
+      try {
+        const result = await update();
+        if (hasPremium(result?.user)) {
+          window.location.href =
+            "http://localhost:6060/redesign/user/dashboard/fix/data-broker-profiles/welcome-to-premium";
+        }
+      } catch (ex) {
+        console.error(ex);
+        captureException(ex);
+      }
+      window.location.href = "/";
+    }
+    void updateSession();
+    // We only want to run this a single time - the page should not be re-rendered.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I think you should be able to have the recommended dependencies here - [the docs do so too](https://next-auth.js.org/getting-started/client#refetching-the-session), so I think `update` only changes when the user e.g. logs out.

```suggestion
  }, [update]);
```

Reply:
Hm I removed this because IIRC it was not working correctly, let me test and report on what happened. I only want this effect to run a single time and not change when properties change in this case FWIW - either we update and the user has a subscription and we redirect them to the welcome page, or something in that chain fails or is false and we redirect them to /.

Reply:
OK it works fine, it does run twice though... I don't understand `useEffect` enough to know why :(

Reply:
Oh yeah that's fine, that's a somewhat contentious React feature if you enable strict mode (which we do, at the moment): [it will render everything twice](https://react.dev/reference/react/StrictMode#fixing-bugs-found-by-double-rendering-in-development) to surface errors resulting from components being impure functions. Strict mode only affects dev mode.